### ISSUE TITLE
Enhance Data Structures: Support for iteration of `dict`

### ISSUE THREAD

**Issue by faze-geek** [lcompilers/lpython#1881]
```
def f():
    d:dict[i32,i32] = {1 : 100,2 : 200,3 : 300,4 : 400}
    i:i32
    for i in d:      # for in works with string, list, tuple, set
        print(i) 
f()

(lp) C:\Users\kunni\lpython>python try.py
1
2
3
4
```

**Comment by czgdp1807** [comment]
First we have to implement `dict.keys` and `dict.values` (1 PR). And then we can implement `for in` for `dict` (and then a separate PR).

**Comment by kabra1110** [comment]
> First we have to implement `dict.keys` and `dict.values` (1 PR)

Should `dict.keys` be a `list`? CPython uses [view objects](https://docs.python.org/3.7/library/stdtypes.html#dictionary-view-objects), which do not support many `list` operations.

**Comment by czgdp1807** [comment]
That’s true. For now let’s keep it as a list. Later if needed we can create a separate type for `dict.keys`.
